compiler/luci/pass/src/RemoveNoEffectSlicePass.cpp

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/RemoveNoEffectSlicePass.h"
+
+#include <luci/IR/CircleNodes.h>
+
+namespace
+{
+
+template <loco::DataType DT>
+bool check_input_output_shape(luci::CircleNode *input, luci::CircleConst *begin,
+                              luci::CircleConst *size)
```

Comment:
@struss Not checking if param is `nullptr` or not might be problem. Static analysis tool will be warning as well. It might as well move below `nullptr` check to here.

Reply:
um... I will update this.